bmc/power.go

```diff
@@ -49,14 +51,15 @@ func setPowerState(ctx context.Context, timeout time.Duration, state string, p [
 		case <-ctx.Done():
 			err = multierror.Append(err, ctx.Err())
 
-			return false, metadata, err
+			return false, metadataLocal, err
 		default:
 			metadataLocal.ProvidersAttempted = append(metadataLocal.ProvidersAttempted, elem.name)
 			ctx, cancel := context.WithTimeout(ctx, timeout)
 			defer cancel()
 			ok, setErr := elem.powerSetter.PowerSet(ctx, state)
 			if setErr != nil {
 				err = multierror.Append(err, errors.WithMessagef(setErr, "provider: %v", elem.name))
+				metadataLocal.FailedConnDetail[elem.name] = setErr.Error()
```

Comment:
Since the error here could be unrelated to the connection Open/Close - maybe we don't include them in `FailedConnDetail` ?

Reply:
metadata is specific to each method call. So a call to open and the power dont have metadata from each other.

Reply:
Okay, since this map was called `FailedConnDetail` - I assumed this would only contain errors specific to the connection setup/teardown. But seems like you'd like to include any sort of failure errors in there?

Reply:
Yeah. Maybe I should change the name. Maybe `FailedMethodDetail` or `FailedProviderDetail`?

Reply:
Thanks, @joelrebel. Updated.